src/main/scala/viper/server/ViperServerProtocol.scala

```diff
@@ -10,12 +10,18 @@ package viper.server
 
 object ViperServerProtocol {
 
-  // Main Actor requests Verification
-  case class Verify(args: List[String])
+  // Main Actor requests Verification with File Name
+  case class Verify(args: List[String],
+                    reporter: Option[viper.silver.reporter.Reporter],
```

Comment:
I think we specified in the interface that a provides their own reporter.
For frontends, this is useful because we can hook up further transformation directly to the reporter.

Reply:
Okay. But then, why is it optional? What is a use case for running the verification with no reporter at all?

Reply:
No reporter is given to the verify method when ViperCoreServer is run in HTTP mode. This mode then simply uses the Result Stream.

Reply:
For frontends we identified two use cases: (1) Use the reporter and nothing else and (2) do not use the reporter, but wait for the results to arrive (via future, promise, or whatever). 
The use case depends on whether or not result streaming is supported.

Reply:
Thanks for explaining the use cases again. I think both of them can be expressed via the reporter interface. ReporterActor is the actor that is responsible for dealing with messages from VerificationWorker. Hence, I think a cleaner solution for the second use case you named is to extend ReporterActor to, e.g., GobraReporterActor, containing the promise object, and override the ```receive``` method such that if finalises the promise of a result. 

Essentially, I see that currently there are two mechanisms: 

* ActorReporter->ReporterActor that put messages into their verification job's queue
* ActorReporter->GobraReporter that maps verification results and finalises the promises, but the promises are stored an optional fields of the Verify message. 

I think the first mechanism strictly supersedes the second one, so I'm wondering if it's really necessary or if it's Gobra-specific, and that we could marry ViperCoreServer and Gobra on a slightly higher abstraction level.

Reply:
What does it mean when you write "put the messages into their verification job's queue". In the interface we agreed on, there is no "verification job's queue", instead there are reporters. The backends (silicon, carbon) use reporter, we want to supply these reporters, so that we do not have to poll on results.

Reply:
Just to clarify again. The reporter is of type option to also cover the case of ViperServer operating in HTTP mode (as before). In this case no reporter is given to the verify method, thus the reporter passed to the case class Verify only exists in ViperCoreServer but not in ViperServerHttp.